packages/kit/src/lib/HookSystem.ts

```diff
@@ -120,17 +175,27 @@ export class HookSystem<
 	}
 
 	async callHook<TName extends Extract<keyof THookFns, string>>(
-		name: TName,
+		nameOrNameAndHookID: TName | { name: TName; hookID: number },
 		...args: Parameters<THookFns[TName]>
 	): Promise<{
 		data: Awaited<ReturnType<THookFns[TName]>>[];
 		errors: HookError[];
 	}> {
-		const hooks = this._registeredHooks[name] ?? [];
+		const hooks: RegisteredHook<THookFns[TName]>[] = [];
+
+		if (typeof nameOrNameAndHookID === "string") {
+			hooks.push(...(this._registeredHooks[nameOrNameAndHookID] ?? []));
+		} else {
+			hooks.push(
+				...(this._registeredHooks[nameOrNameAndHookID.name]?.filter(
+					(hook) => hook.meta.id === nameOrNameAndHookID.hookID,
+				) ?? []),
+			);
+		}
```

Comment:
I think we could simplify this logic. We could also throw an error if a hook could not be found by ID.

```typescript
let hooks: RegisteredHook<THookFns[TName]>[];

if (typeof nameOrNameAndHookID === "string") {
	hooks = this._registeredHooks[nameOrNameAndHookID] ?? [];
} else {
	const hookForID = this._registeredHooks[nameOrNameAndHookID.name].find(
		(hook) => hook.meta.id === nameOrNameAndHookID.hookID,
	);

	if (hookForID) {
		hooks = [hookForID];
	} else {
		// Should we throw an error here?
	}
}
```

Reply:
Right, we talked about that!